Error on update to service settings from http3 validation

With the following plan, and HTTP3 already enabled:

```hcl
 # module.user-id-svc-test.fastly_service_vcl.this will be updated in-place
  ~ resource "fastly_service_vcl" "this" {
      ~ active_version     = 51 -> (known after apply)
      ~ cloned_version     = 51 -> (known after apply)
      ~ default_ttl        = 0 -> 3600
        id                 = "2sTxhg5fdZLa6aBWbjGc6g"
        name               = "..."
        # (7 unchanged attributes hidden)

        # (30 unchanged blocks hidden)
    }
```

An invalid request is made to the Fastly API resulting in a 400 error and a failed apply:

```
2023-03-29T13:43:49.400+0100 [INFO]  provider: configuring client automatic mTLS
2023-03-29T13:43:49.407+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/fastly/fastly/3.2.0/darwin_arm64/terraform-provider-fastly_v3.2.0 args=[.terraform/providers/registry.terraform.io/fastly/fastly/3.2.0/darwin_arm64/terraform-provider-fastly_v3.2.0]
2023-03-29T13:43:49.410+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/fastly/fastly/3.2.0/darwin_arm64/terraform-provider-fastly_v3.2.0 pid=19236
2023-03-29T13:43:49.410+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/fastly/fastly/3.2.0/darwin_arm64/terraform-provider-fastly_v3.2.0
2023-03-29T13:43:49.421+0100 [INFO]  provider.terraform-provider-fastly_v3.2.0: configuring server automatic mTLS: timestamp=2023-03-29T13:43:49.420+0100
2023-03-29T13:43:49.436+0100 [DEBUG] provider: using plugin: version=5
2023-03-29T13:43:49.436+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: plugin address: address=/var/folders/nz/wb_0py2x44qclwdnpwvbqtlr0000gp/T/plugin1663656525 network=unix timestamp=2023-03-29T13:43:49.436+0100
module.user-id-svc-test.fastly_service_vcl.this: Modifying... [id=2sTxhg5fdZLa6aBWbjGc6g]
module.user-id-svc-prod.fastly_service_vcl.this: Modifying... [id=3qCGuhVjMpmv0Zh3kcXPXi]
2023-03-29T13:43:49.610+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: Creating clone of version (51) for updates: timestamp=2023-03-29T13:43:49.610+0100
2023-03-29T13:43:49.616+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: Creating clone of version (56) for updates: timestamp=2023-03-29T13:43:49.616+0100
2023-03-29T13:43:49.972+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: Sleeping 7 seconds to allow Fastly Version to be available: timestamp=2023-03-29T13:43:49.972+0100
2023-03-29T13:43:50.346+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: Sleeping 7 seconds to allow Fastly Version to be available: timestamp=2023-03-29T13:43:50.346+0100
2023-03-29T13:43:56.973+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: Update Settings opts: fastly.UpdateSettingsInput{DefaultHost:(*string)(0x14000774450), DefaultTTL:0xe10, ServiceID:"2sTxhg5fdZLa6aBWbjGc6g", ServiceVersion:55, StaleIfError:(*bool)(0x14000514128), StaleIfErrorTTL:(*uint)(0x14000514110)}: timestamp=2023-03-29T13:43:56.973+0100
2023-03-29T13:43:57.347+0100 [DEBUG] provider.terraform-provider-fastly_v3.2.0: Update Settings opts: fastly.UpdateSettingsInput{DefaultHost:(*string)(0x14000284730), DefaultTTL:0xe10, ServiceID:"3qCGuhVjMpmv0Zh3kcXPXi", ServiceVersion:60, StaleIfError:(*bool)(0x140007b2288), StaleIfErrorTTL:(*uint)(0x140007b2270)}: timestamp=2023-03-29T13:43:57.347+0100
2023-03-29T13:43:57.421+0100 [ERROR] provider.terraform-provider-fastly_v3.2.0: Response contains error diagnostic: diagnostic_summary="400 - Bad Request:

    Title:  Bad request
    Detail: Count can't exceed 1 HTTP3/QUIC module per service" tf_resource_type=fastly_service_vcl @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR tf_proto_version=5.3 tf_provider_addr=provider @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 tf_rpc=ApplyResourceChange tf_req_id=08de5dbc-cb12-defb-a2fe-d2c53e04b655 timestamp=2023-03-29T13:43:57.420+0100
2023-03-29T13:43:57.864+0100 [ERROR] provider.terraform-provider-fastly_v3.2.0: Response contains error diagnostic: tf_resource_type=fastly_service_vcl @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_detail= diagnostic_severity=ERROR tf_provider_addr=provider tf_req_id=827ba095-8d7c-63d9-85aa-b7316833d5a1 tf_rpc=ApplyResourceChange @module=sdk.proto diagnostic_summary="400 - Bad Request:

    Title:  Bad request
    Detail: Count can't exceed 1 HTTP3/QUIC module per service" tf_proto_version=5.3 timestamp=2023-03-29T13:43:57.864+0100
╷
│ Error: 400 - Bad Request:
│ 
│     Title:  Bad request
│     Detail: Count can't exceed 1 HTTP3/QUIC module per service
│ 
│   with module.user-id-svc-prod.fastly_service_vcl.this,
│   on .terraform/modules/user-id-svc-prod/terraform/modules/services/multi-origin-service/main.tf line 89, in resource "fastly_service_vcl" "this":
│   89: resource "fastly_service_vcl" "this" {
│ 
╵
╷
│ Error: 400 - Bad Request:
│ 
│     Title:  Bad request
│     Detail: Count can't exceed 1 HTTP3/QUIC module per service
│ 
│   with module.user-id-svc-test.fastly_service_vcl.this,
│   on .terraform/modules/user-id-svc-test/terraform/modules/services/multi-origin-service/main.tf line 89, in resource "fastly_service_vcl" "this":
│   89: resource "fastly_service_vcl" "this" {
│ 
╵
Releasing state lock. This may take a few moments...
2023-03-29T13:43:59.871+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-03-29T13:43:59.873+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/fastly/fastly/3.2.0/darwin_arm64/terraform-provider-fastly_v3.2.0 pid=19236
2023-03-29T13:43:59.873+0100 [DEBUG] provider: plugin exited
```

I'll get around this by manually activating the draft for now.

Relates to https://github.com/fastly/terraform-provider-fastly/pull/640.

https://github.com/fastly/terraform-provider-fastly/blob/c0268a92b4c6147f3edc0e280046cd8cc76044da/fastly/block_fastly_service_settings.go#L42-L55

--- a/fastly/block_fastly_service_settings.go
+++ b/fastly/block_fastly_service_settings.go
@@ -41,11 +41,22 @@ func (h *SettingsServiceAttributeHandler) Process(_ context.Context, d *schema.R
 
 	if attr, ok := d.GetOk("http3"); ok {
 		if attr.(bool) {
-			_, err = conn.EnableHTTP3(&gofastly.EnableHTTP3Input{
-				FeatureRevision: gofastly.Int(1),
-				ServiceID:       d.Id(),
-				ServiceVersion:  latestVersion,
-			})
+			// IMPORTANT: API will 400 when trying to enable HTTP3 when already on.
+			//
+			// So we first check the HTTP3 status.
+			// The API returns a 404 if HTTP3 is not enabled.
+			// The API client returns an error for non-2xx responses.
+			// So if there is no error, then HTTP3 is enabled.
+			if _, err = conn.GetHTTP3(&gofastly.GetHTTP3Input{
+				ServiceID:      d.Id(),
+				ServiceVersion: latestVersion,
+			}); err != nil {
+				_, err = conn.EnableHTTP3(&gofastly.EnableHTTP3Input{
+					FeatureRevision: gofastly.Int(1),
+					ServiceID:       d.Id(),
+					ServiceVersion:  latestVersion,
+				})
+			}
 		}
 	} else {
 		err = conn.DisableHTTP3(&gofastly.DisableHTTP3Input{

